client/tx/tx.go

```diff
@@ -418,18 +419,25 @@ func Sign(txf Factory, name string, txBuilder client.TxBuilder) error {
 		Data:     &sigData,
 		Sequence: txf.Sequence(),
 	}
+	var prevSignatures []signing.SignatureV2
+	if overwriteSig {
+		prevSignatures, err = txBuilder.GetTx().GetSignaturesV2()
+		if err != nil {
+			return err
+		}
+	}
 	if err := txBuilder.SetSignatures(sig); err != nil {
```

Comment:
There may be something slightly more complicated going on here that we need to be careful about. If you read above, we are only calling this `SetSignatures()` method with an empty sig so that the underlying `txBuilder` can set the `SignModes` which is necessary for generating the bytesToSign below.

In the case of `--append` flag, I think that we actually want to call `SetSignatures()` with the pre-existing list of existing signatures + the empty new signature.

e.g. i think the full overwrite logic here needs to be something more like:

```go
	if overwriteSig {
        	if err := txBuilder.SetSignatures(sig); err != nil {
	        	return err
	        }
	} else {
		prevSignatures, err = txBuilder.GetTx().GetSignaturesV2()
		if err != nil {
			return err
		}
                sigs := append(prevSignatures, sig)
        	if err := txBuilder.SetSignatures(sigs); err != nil {
	        	return err
	        }
        }
```

Would be good to get confirmation from @amaurymartiny but i think this is correct logic.

Reply:
I've added a test to verify the matching signatures. It works. The reason the signatures are reset here is to set the right signing mode. 

The `wrapper` and client signing functions are not very clear. As I wrote in the summary above, it should probably be rewritten.

Reply:
I think the edge case that might be missing here is if you have two different signatures with different sign modes. Does the test work in that case as well?

Reply:
I've added one more tests for signing with a different factory which has a different mode:
```
{"should succeed to append a signature with different mode",
	txfAmino, from1, false, []cryptotypes.PubKey{pubKey2, pubKey1}, []int{0, 0}},
```

In fact your patch doesn't work, because the `wrapper` (TxBuilder) is wrongly implemented - it doesn't know which key it should use to sign nor the mode to use. That logic is handled independently, through `txFactory` and maybe totally different then the mode set in the `TxBuilder` signature placeholder. In my opinion this is error prone, but that's something for a new task.